docs/core/diagnostics/trace-perfcollect-lttng.md

```diff
@@ -0,0 +1,201 @@
+ms.date: 10/23/2020
+---
+
+# Tracing .NET applications with LTTng and PerfCollect
+
+**This article applies to: ✔️** .NET Core 2.1 SDK and later versions
+
+When a performance problem is encountered on Linux, collecting a trace with `perfcollect` can be used to gather detailed information about what was happening on the machine at the time of the performance problem.
+
+`perfcollect` is a bash script that leverages [Linux Tracing Tookit-Next Generation (LTTng)](https://lttng.org) to collect events written from the runtime or any [EventSource](xref:System.Diagnostics.Tracing.EventListener), as well as [perf](https://perf.wiki.kernel.org/) to collect CPU samples of the target process.
+
+## Preparing Your Machine
+
+Follow these steps to prepare your machine to collect a performance trace with `perfcollect`.
+
+1. Download `perfcollect`.
+
+    > ```bash
+    > curl -OL http://aka.ms/perfcollect
+    > ```
+
+2. Make the script executable.
+
+    > ```bash
+    > chmod +x perfcollect
+    > ```
+
+3. Install tracing prerequisites - these are the actual tracing libraries.
+
+    > ```bash
+    > sudo ./perfcollect install
+    > ```
+
+    This will install the following prerequisites on your machine:
+
+    1. `perf`: the Linux Performance Events sub-system and companion user-mode collection/viewer application. `perf` is part of the Linux kernel source, but is not usually installed by default.
+
+    2. `LTTng`: Used to capture event data emitted at runtime by CoreCLR. This data is then used to analyze the behavior of various runtime components such as the GC, JIT and thread pool.
+
+## Collecting a Trace
+
+1. Have two shells available - one for controlling tracing, referred to as **[Trace]**, and one for running the application, referred to as **[App]**.
+
+2. **[App]** Setup the application shell with the following environment variables - this enables tracing configuration of CoreCLR.
+
+    > ```bash
+    > export COMPlus_PerfMapEnabled=1
+    > export COMPlus_EnableEventLog=1
+    > ```
+
+3. **[Trace]** Start collection.
+
+    > ```bash
+    > sudo ./perfcollect collect sampleTrace
+    > ```
+
+    Expected Output:
+
+    > ```bash
+    > Collection started.  Press CTRL+C to stop.
+    > ```
+
+4. **[App]** Run the app - let it run as long as you need to in order to capture the performance problem.
+
+    > ```bash
+    > dotnet run
+    > ```
+
+5. **[Trace]** Stop collection - hit CTRL+C.
+
+    > ```bash
+    > ^C
+    > ...STOPPED.
+    >
+    > Starting post-processing. This may take some time.
+    >
+    > Generating native image symbol files
+    > ...SKIPPED
+    > Saving native symbols
+    > ...FINISHED
+    > Exporting perf.data file
+    > ...FINISHED
+    > Compressing trace files
+    > ...FINISHED
+    > Cleaning up artifacts
+    > ...FINISHED
+    >
+    > Trace saved to sampleTrace.trace.zip
+    > ```
+
+    The compressed trace file is now stored in the current working directory.
+
+## Resolving Framework Symbols
```

Comment:
@sdmaclea @brianrob - Am I remembering correctly that this step is no longer required as long as the developer is tracing .Net Core 3.0+ (which has Steve's perf jitdump changes) and they have a sufficiently recent version of perf? Do we have a good description of what environments will have the updated version of perf or is this something that the perfcollect could warn about during the install step?

What I am hoping we can do is insert a tiny piece of text in the workflow above that says something like:
````
Recent versions of .NET Core and the Linux perf tool support automatic resolution of method names for 
framework code. If you are working with .NET Core version 2.1 or perf version <= XXX an extra step is
 necessary. See the [instructions below](some_anchor_link) for details.
````

Reply:
Yup, but it's .NET 5.  So starting with .NET 5, you no longer need to do anything in this section to resolve framework symbols.